internal/controller/sync.go

```diff
@@ -294,15 +298,22 @@ func (r *Route) createNextCR(ctx context.Context, route *routev1.Route, revision
 		return nil
 	}
 
-	// Parse out SANs
 	var dnsNames []string
+	// Get the canonical hostname(s) of the Route (from .spec.host or .spec.subdomain)
+	dnsNames = getRouteHostnames(route)
+	if len(dnsNames) <= 0 {
```

Comment:
Are negative integers possible values returned by `getRouteHostnames`? If not, I'd avoid the ambiguity created by `<=`:

```suggestion
	if len(dnsNames) == 0 {
```